recipes/gmp/all/conanfile.py

```diff
@@ -36,46 +42,79 @@ def package_id(self):
+            configure_file = os.path.join(self._source_subfolder, "configure")
+            tools.replace_in_file(configure_file, r"-install_name \$rpath/", "-install_name ")
+            configure_stats = os.stat(configure_file)
+            os.chmod(configure_file, configure_stats.st_mode | stat.S_IEXEC)
+        yes_no = lambda v: "yes" if v else "no"
+        configure_args = [
+            "--enable-assembly={}".format(yes_no(not self.options.disable_assembly)),
+            "--enable-cxx={}".format(yes_no(self.options.enable_cxx)),
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--srcdir={}".format(os.path.join(self.source_folder, self._source_subfolder).replace("\\", "/")),
+        ]
+        if self.settings.compiler == "Visual Studio":
+            configure_args.extend([
+                "ac_cv_c_restrict=restrict",
+                "gmp_cv_asm_label_suffix=:",
+                "lt_cv_sys_global_symbol_pipe=cat",  # added to get further in shared MSVC build, but it gets stuck later
+            ])
+            self._autotools.flags.append("-FS")
+            # self._autotools.cxx_flags.append("-EHsc")
+        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
         return self._autotools
 
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                yasm_machine = {
+                    "x86": "x86",
+                    "x86_64": "amd64",
+                }[str(self.settings.arch)]
+                env = {
+                    "CC": "cl -nologo",
+                    "CCAS": "{} -a x86 -m {} -p gas -r raw -f win32 -g null -X gnu".format(os.path.join(self.build_folder, "yasm_wrapper.sh").replace("\\", "/"), yasm_machine),
+                    "CXX": "cl -nologo",
+                    "AR": "{} lib".format(self.deps_user_info["automake"].ar_lib.replace("\\", "/")),
+                    "LD": "link -nologo",
+                    "NM": "python {}/{}".format(self.build_folder.replace("\\", "/"), "dumpbin_nm.py"),
+                }
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
     def build(self):
-        with tools.chdir(self._source_subfolder):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        with self._build_context():
             autotools = self._configure_autotools()
             autotools.make()
-        # INFO: According to the gmp readme file, make check should not be omitted, but it causes timeouts on the CI server.
-        if self.options.run_checks:
-            autotools.make(args=["check"])
+            # INFO: According to the gmp readme file, make check should not be omitted, but it causes timeouts on the CI server.
```

Comment:
🏴

Reply:
I think this is correct.
The tests are run in the build context.
The checks are also disabled by default, so there will be no timeouts.